module/apmhttp/handler_test.go

```diff
@@ -107,6 +108,36 @@ func TestHandler(t *testing.T) {
 	}, transaction.Context)
 }
 
+func TestCaptureErrorWithRequestBody(t *testing.T) {
+	tracer, transport := transporttest.NewRecorderTracer()
+	defer tracer.Close()
+	tracer.SetCaptureBody(apm.CaptureBodyAll)
+
+	mux := http.NewServeMux()
+	mux.Handle("/foo", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ioutil.ReadAll(req.Body)
+		w.WriteHeader(http.StatusServiceUnavailable)
+		errorMsg := "total explosion"
+		e := apm.CaptureError(req.Context(), errors.New(errorMsg))
+		e.Send()
+		w.Write([]byte(errorMsg))
+	}))
+
+	h := apmhttp.Wrap(mux, apmhttp.WithTracer(tracer))
+	w := httptest.NewRecorder()
+
+	req, _ := http.NewRequest("POST", "http://server.testing/foo", strings.NewReader("foo-body"))
+	req.Header.Set("User-Agent", "apmhttp_test")
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.RemoteAddr = "client.testing:1234"
+	h.ServeHTTP(w, req)
+	tracer.Flush(nil)
+
+	payloads := transport.Payloads()
+	e := payloads.Errors[0]
+	assert.Equal(t, &model.RequestBody{Raw: "foo-body"}, e.Context.Request.Body)
```

Comment:
We should also check that the transaction still has the request body.

Reply:
Also, can you please add this new test to `apmtest.HTTPTestSuite` instead? This functionality should be common across each of the HTTP integrations. If the implementation was all inside `apmhttp.StartTransaction` then it would be fine to test in one place.